.ci/run-elasticsearch.sh

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
+#
+# Launch one or more Elasticsearch nodes via the Docker image,
+# to form a cluster suitable for running the REST API tests.
+#
+# Export the ELASTICSEARCH_VERSION variable, eg. 'elasticsearch:8.0.0-SNAPSHOT'.
+
+# Version 1.0.2
```

Comment:
this file should not live in this repo if it is shared by multiple clients, instead it should be in the shared repository somewhere to avoid duplication. It is also unnecessarily complex for the needs of the tests. To run the server for x-pack tests all you need is:

```
docker run \
    --publish 9200:9200 \
    --env "xpack.security.authc.token.enabled=true" \
    --env "xpack.security.authc.api_key.enabled=true" \
    --env "xpack.security.enabled=true"  \
    --env "ELASTIC_PASSWORD=changeme" \
    --env "xpack.license.self_generated.type=trial" \
    --env "discovery.type=single-node" \
    --env "path.repo=/tmp" \
    --env "repositories.url.allowed_urls=http://*" \
    --env "node.attr.testattr=test"  \
    --env "node.name=test"  \
    docker.elastic.co/elasticsearch/elasticsearch:$VERSION
```
(this does not include SSL which can bea easily added).

For non x-pack tests it is
```
docker run --detach --rm --publish 9200:9200 --env "discovery.type=single-node" -e path.repo=/tmp -e "repositories.url.allowed_urls=http://*" -e node.attr.testattr=test  -e node.name=test  docker.elastic.co/elasticsearch/elasticsearch:$VERSION
```

not sure why this entire infrastructure is needed and what it adds for the complexity.

Reply:
For now this does need to live in this repository as well. The added complexity is in part to support running in different modes. e.g all the clients can run the yaml tests using:

```
$ ELASTICSEARCH_VERSION=7.x-SNAPSHOT TEST_SUITE=xpack ./.ci/run-tests 
```

Or choose to run `run-elasticsearch` separately and (using DETACH=true or not) during development. 

After we get everyone in this harness we'll explore our options to move parts of this into a shared space.